app/policies/theblog/content_node_policy.rb

```diff
@@ -25,7 +25,7 @@ def unblock?
     end
 
     def account_can_update_record?
-      (user == record.author && account_has_role?(user, [:editor])) ||
+      (user.try(:id) == record.author.try(:id) && account_has_role?(user, [:editor])) ||
```

Comment:
about changing relation - it was the first thing came to my mind, but once we still use Incarnator.account_model to authorise, there are a lot of other troubles.
About has_role? - could you explain?